### PR TITLE
Enabled load model from buffer for rust

### PR DESCRIPTION
**Pull request by manavsah**
Before submitting a pull request, please do the following steps:

1. Read [instructions for contributors](https://catboost.ai/docs/concepts/development-and-contributions.html).
2. Run `ya make` in catboost folder to make sure the code builds.
3. Add tests that test your change.
4. Run tests using `ya make -t -A` command.
5. If you haven't already, complete the [CLA](https://yandex.ru/legal/cla/).

### CODE REVIEW

**Comment by kizill** [comment]
Hello! Thank you for PR! We've recently done something that could be useful - non-owning model deserialization. When you have some fixed buffer or memory region with model data (for example, memory mapped file) you can deserialize model with nearly zero-cost in terms of memory allocations. Would it be usefull for you?

File: .gitignore [hunk 1]
```diff
@@ -25,3 +25,20 @@ test-results
...
 /msvs/Debug/*
 /msvs/Release/*
+
+catboost/rust-package/target/
+
+catboost/rust-package/catboost-sys/target/
+
+catboost/app/catboost
+
+catboost/libs/model_interface/libcatboostmodel.dylib
```
Comment (kizill) [inline .gitignore:35]: i suggest to use more generic `*.dylib*` instead of exact path to artifact.

**Review by kizill — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (manavsah) [reply .gitignore:35]: thanks will update

**Comment by kizill** [comment]
@arcadia-devtools Ship it!

**Comment by manavsah** [comment]
> Hello! Thank you for PR! We've recently done something that could be useful - non-owning model deserialization. When you have some fixed buffer or memory region with model data (for example, memory mapped file) you can deserialize model with nearly zero-cost in terms of memory allocations. Would it be usefull for you?

i am using "LoadFullModelFromBuffer" function  declared in catboost/catboost/libs/model_interface/c_api.h.
LoadFullModelFromBuffer itself is calling your updated function "NCB::IModelLoader::ReadModel".


Please  correct me if needed.

CATBOOST_API bool LoadFullModelFromBuffer(
    ModelCalcerHandle* modelHandle,
    const void* binaryBuffer,
    size_t binaryBufferSize);

**Comment by arcadia-devtools** [comment]
CLA already signed

**Comment by arcadia-devtools** [comment]
@kizill, internal review request created: [1521050](https://a.yandex-team.ru/review/1521050)

**Comment by arcadia-devtools** [comment]
Check [1521050](https://a.yandex-team.ru/review/1521050) passed

**Comment by leofidus** [comment]
What is the status of this? Is there anything we can do to move it along?

**Comment by manavsah** [comment]
Hi,

Changes are complete for this PR. 
I am also waiting for them to be reviewed & accepted.

Thanks

**Comment by kizill** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
@kizill, internal review request updated: [1521050](https://a.yandex-team.ru/review/1521050)

**Review by kizill — APPROVED** [review]

**Comment by manavsah** [comment]
Thanks

**Pull request merged**